src/SignalR/clients/cpp/src/signalrclient/hub_connection.cpp

```diff
@@ -37,24 +37,26 @@ namespace signalr
         return m_pImpl->on(event_name, handler);
     }
 
-    pplx::task<web::json::value> hub_connection::invoke(const std::string& method_name, const web::json::value& arguments)
+    void hub_connection::invoke(const std::string& method_name, const web::json::value& arguments, std::function<void(const web::json::value&, std::exception_ptr)> callback) noexcept
     {
         if (!m_pImpl)
         {
-            throw signalr_exception("invoke() cannot be called on uninitialized hub_connection instance");
+            callback(web::json::value(), std::make_exception_ptr(signalr_exception("invoke() cannot be called on uninitialized hub_connection instance")));
```

Comment:
Hmm this exception message feels a little weird. You need to say that the destructor has been called already